crates/sui-graphql-rpc/src/extensions/logger.rs

```diff
@@ -41,18 +44,39 @@ pub struct Logger {
 impl ExtensionFactory for Logger {
     fn create(&self) -> Arc<dyn Extension> {
         Arc::new(LoggerExtension {
+            query: "".to_string().into(),
+            query_id: "".to_string().into(),
             session_id: "".to_string().into(),
             config: self.config.clone(),
         })
     }
 }
 
 struct LoggerExtension {
-    session_id: Mutex<String>,
     config: LoggerConfig,
+    query: Mutex<String>,
+    query_id: Mutex<String>,
+    session_id: Mutex<String>,
 }
 
 impl LoggerExtension {
+    async fn set_query(&self, query: &str) {
+        *self.query.lock().await = query.to_string();
+    }
+    async fn query(&self) -> String {
```

Comment:
I tried to return a `&str` instead of `String` but couldn't figure out the lifetimes.